create.correlateUpdateMailboxRegionalConfigurationAndFolderPermission.ps1

```diff
@@ -88,630 +74,303 @@ function Resolve-HTTPError {
+        $auditLogs.Add([PSCustomObject]@{
+                # Action  = "" # Optional
+                Message = "Error connecting to Exchange Online. Error Message: $($errorMessage.AuditErrorMessage)"
+                IsError = $True
+            })
+
+        # Skip further actions, as this is a critical error
+        continue
+    }
+
+    # Get Exchange Online Mailbox
+    try {
+        Write-Verbose "Querying EXO mailbox where [$($correlationProperty)] = [$($correlationValue)]"
+            
+        $mailbox = Get-EXOMailbox -Filter "$($correlationProperty) -eq '$($correlationValue)'" -ErrorAction Stop
+
+        if (($mailbox | Measure-Object).Count -eq 0) {
+            throw "Could not find a EXO mailbox where [$($correlationProperty)] = [$($correlationValue)]" 
         }
-        finally {
-            $aRef = $getExoMailbox.aRef
-            $auditLogs += $getExoMailbox.auditLogs
-            $mailbox = $getExoMailbox.mailbox
-
-            # Log the data from logging arrays (since the "normal" Write-Information isn't sent to HelloID as another PS session performs the commands)
-            $verboseLogs = $getExoMailbox.verboseLogs
-            foreach ($verboseLog in $verboseLogs) { Write-Verbose $verboseLog }
-            $informationLogs = $getExoMailbox.informationLogs
-            foreach ($informationLog in $informationLogs) { Write-Information $informationLog }
-            $warningLogs = $getExoMailbox.warningLogs
-            foreach ($warningLog in $warningLogs) { Write-Warning $warningLog }
+
+        # Set aRef object for use in futher actions
+        $aRef = [PSCustomObject]@{
+            Guid              = $mailbox.Guid
+            UserPrincipalName = $mailbox.UserPrincipalName
         }
+
+        $auditLogs.Add([PSCustomObject]@{
+                # Action  = "" # Optional
+                Message = "Successfully queried and correlated to EXO mailbox [$($aRef.userPrincipalName) ($($aRef.Guid))]"
+                IsError = $false
+            })
     }
+    catch { 
+        $ex = $PSItem
+        $errorMessage = Get-ErrorMessage -ErrorObject $ex
+
+        Write-Verbose "Error at Line [$($ex.InvocationInfo.ScriptLineNumber)]: $($ex.InvocationInfo.Line). Error: $($errorMessage.VerboseErrorMessage)"
+        $auditLogs.Add([PSCustomObject]@{
+                # Action  = "" # Optional
+                Message = "Error querying EXO mailbox where [$($correlationProperty)] = [$($correlationValue)]. Error Message: $($errorMessage.AuditErrorMessage)"
+                IsError = $True
+            })
+    }
+
+    # Update Mailbox Folder Permission
+    try {
+        Write-Verbose "Updating folder permissions for mailbox [$($aRef.userPrincipalName) ($($aRef.Guid))]: $($mailboxSplatParams | ConvertTo-Json)"
+
+        # Get Mailbox "Calendar" folder name
+        $mailboxFolderId = Get-MailboxFolderStatistics -Identity $mailbox.Guid -FolderScope Calendar | Where-Object { $_.FolderType -eq 'Calendar' } | Select-Object Name
```

Comment:
I get this result: @{Name=Agenda}

Suggestion:
(Get-MailboxFolderStatistics -Identity $mailbox.Guid -FolderScope Calendar | Where-Object { $_.FolderType -eq 'Calendar' } | Select-Object Name).Name

Result: Agenda

Reply:
Good catch. corrected